src/main/java/com/hotels/mutantswarm/MutantSwarmExtension.java

```diff
@@ -0,0 +1,179 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.mutantswarm;
+
+import static org.reflections.ReflectionUtils.withAnnotation;
+
+import java.lang.reflect.Field;
+import java.nio.charset.Charset;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.extension.AfterAllCallback;
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.TestInstancePostProcessor;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContext;
+import org.junit.jupiter.api.extension.TestTemplateInvocationContextProvider;
+import org.junit.jupiter.api.extension.TestWatcher;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.Preconditions;
+import com.klarna.hiverunner.HiveRunnerExtension;
+import com.klarna.hiverunner.HiveShellContainer;
+import com.klarna.hiverunner.annotations.HiveSQL;
+import com.klarna.hiverunner.builder.HiveShellBuilder;
+import com.klarna.hiverunner.builder.Script;
+import com.klarna.hiverunner.config.HiveRunnerConfig;
+import com.klarna.reflection.ReflectionUtils;
+
+import com.hotels.mutantswarm.MutantSwarmCore.ExecutionContext;
+import com.hotels.mutantswarm.exec.MutantState;
+import com.hotels.mutantswarm.exec.MutatedSourceFactory.MutatedSource;
+import com.hotels.mutantswarm.exec.SwarmResults;
+import com.hotels.mutantswarm.exec.SwarmResults.SwarmResultsBuilder;
+import com.hotels.mutantswarm.model.MutantSwarmScript;
+import com.hotels.mutantswarm.plan.Mutant;
+import com.hotels.mutantswarm.plan.Swarm;
+import com.hotels.mutantswarm.report.ReportGenerator;
+
+public class MutantSwarmExtension extends HiveRunnerExtension implements AfterAllCallback, TestWatcher,
+    TestTemplateInvocationContextProvider, TestInstancePostProcessor, AfterEachCallback {
+
+  private static final Logger log = LoggerFactory.getLogger(MutantSwarmExtension.class);
+  private static AtomicReference<ExecutionContext> contextRef = new AtomicReference<>();
+
+  public HiveShellContainer container;
+  private HiveRunnerConfig HiveRunnerConfig = new HiveRunnerConfig();
+  private int testNumber = -1;
+  private List<Mutant> mutants = new ArrayList<Mutant>();
+  private List<MutatedSource> mutatedSources = new ArrayList<MutatedSource>();
+  private ExecutionContext resultContext = contextRef.get();
+  private boolean firstTestPassed = true;
+  private MutantSwarmCore core = new MutantSwarmCore();
+
+  public MutantSwarmExtension() {}
```

Comment:
Since there are no other constructors you can leave this one out, the compiler will automatically add a zero-args constructor for you.